src/trans/gpu/algor/cuda_device_mod.F90

```diff
@@ -0,0 +1,75 @@
+! (C) Copyright 2020- ECMWF.
+!
+! This software is licensed under the terms of the Apache Licence Version 2.0
+! which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+! In applying this licence, ECMWF does not waive the privileges and immunities
+! granted to it by virtue of its status as an intergovernmental organisation
+! nor does it submit to any jurisdiction.
+!
+
+module cuda_device_mod
+
+        use cudafor
+interface device_sync
+
+integer function device_synchronize() bind(C,name='cudaDeviceSynchronize')
+use iso_c_binding
+end function device_synchronize
+
+end interface device_sync
+
+interface devicestreamsync
+
+integer function device_stream_synchronize(stream) bind(C,name='cudaStreamSynchronize')
+use iso_c_binding
+type(c_ptr) :: stream
+end function device_stream_synchronize
+
+end interface devicestreamsync
+
+interface devicestreamdestroy
+
+integer function device_stream_destroy(stream) bind(C,name='cudaStreamDestroy')
+use iso_c_binding
+type(c_ptr) :: stream
+end function device_stream_destroy
+
+end interface devicestreamdestroy
+
+interface devicesetdevice
+
+integer function device_SetDevice(devnum) bind(C,name='cudaSetDevice')
+use iso_c_binding
+integer(c_int),value :: devnum
+end function device_SetDevice
+
+end interface devicesetdevice
+
+interface devicegetdevice
+
+integer function device_GetDevice(devnum) bind(C,name='cudaGetDevice')
+use iso_c_binding
+integer(c_int) :: devnum
+end function device_GetDevice
+
+end interface devicegetdevice
+
+interface devicegetdevicecount
+
+integer function device_GetDeviceCount(devnum) bind(C,name='cudaGetDeviceCount')
+use iso_c_binding
+integer(c_int) :: devnum
+end function device_GetDeviceCount
+
+end interface devicegetdevicecount
+
+interface devicegetmeminfo
+
+integer function device_MemGetInfo(memfree_mb,memtotal_mb) bind(C,name='c_hipmemgetinfo')
```

Comment:
I can't see any definition of the function `c_hipmemgetinfo`
In fact there is no reference to use these functions from Fortran either:
`device_MemGetInfo` or `devicegetmeminfo`
It would be good to delete this.
The same goes for these in `hip_device_mod.F90`

Reply:
I actively used the devicememgetinfo implementations when debugging this branch, both on local GPUs and on lumi. But indeed, my implementation got lost in the run up to this merge. 
Personally I would prefer to keep the memgetinfo interface for a rainy day (and therefore add the cpp file below to the PR), even if it's not actively used, but if you prefer to delete it I won't protest. 

The missing cpp code is trivial : 
cat c_hipmemgetinfo.cpp
```
#include <string.h>
#include "hicblas.h"


extern "C" {
hipError_t c_hipmemgetinfo( int *meg_free, int *meg_total)
{

  size_t l_free  = 0;
  size_t l_total = 0;

  hipError_t error_memgetinfo;
  error_memgetinfo = hipMemGetInfo(&l_free, &l_total);

  long long ll_free = (long long) l_free;
  long long ll_total = (long long) l_total;

  *meg_free  = (int) (ll_free  / 1048576);
  *meg_total = (int) (ll_total / 1048576);

  return error_memgetinfo;
}

}
```

Reply:
Aha good to see. It can be added with the previously missing implementation. Is this also working in the cuda_device_mod?

Reply:
Thoughts? Do we delete, or should I add this file to the PR?

Reply:
Should compile as is : the hicblas_cuda.h header defines hipimemgetinfo as cudamemgetinfo

Reply:
Great :)